python/pysrc/kaskada/_execution.py

```diff
@@ -38,6 +41,39 @@ class _ExecutionOptions:
     #: For snapshots, this determines when the snapshot is produced.
     final_at: Optional[int] = None
 
+    @staticmethod
+    def create(
+        results: Optional[Union[kd.results.History, kd.results.Snapshot]],
+        row_limit: Optional[int],
+        max_batch_size: Optional[int],
+        mode: Literal["once", "live"] = "once",
+    ) -> _ExecutionOptions:
+        """Create execution options."""
+        options = _ExecutionOptions(
+            row_limit=row_limit,
+            max_batch_size=max_batch_size,
+            materialize=mode == "live",
+        )
+
+        if results is None:
+            results = kd.results.History()
+
+        if isinstance(results, kd.results.History):
+            options.results = "history"
+            if results.since is not None:
+                options.changed_since = int(results.since.timestamp())
+            if results.until is not None:
+                options.final_at = int(results.until.timestamp())
+        elif isinstance(results, kd.results.Snapshot):
+            options.results = "snapshot"
+            if results.changed_since is not None:
+                options.changed_since = int(results.changed_since.timestamp())
+            if results.at is not None:
+                options.final_at = int(results.at.timestamp())
```

Comment:
Do we want any logic here that asserts `final_at` is greater than `changed_since`?

Reply:
Hm, yeah. Document in the API, then we can enforce it in Sparrow at a single spot before compilation then